engine/gamesys/src/gamesys/gamesys_private.cpp

```diff
@@ -46,13 +46,21 @@ namespace dmGameSystem
             const char* path_name_receiver = dmHashReverseSafe64(receiver->m_Path);
             const char* fragment_name_receiver = dmHashReverseSafe64(receiver->m_Fragment);
 
-            n+= dmSnPrintf(buf + n, sizeof(buf) - n, " Message '%s' sent from %s:%s#%s to %s:%s#%s.",
+            int length = dmSnPrintf(buf + n, sizeof(buf) - n, " Message '%s' sent from %s:%s#%s to %s:%s#%s.",
                             id_str,
                             socket_name_sender, path_name_sender, fragment_name_sender,
                             socket_name_receiver, path_name_receiver, fragment_name_receiver);
+            if (length == -1)
+            {
+                n = sizeof(buf);
```

Comment:
Is this correct?
The string length is `n - 1` here?

Reply:
The length is indeed `n - 1` as a result of truncation.

However, in order for the underflow error below to fire, `n` can't be set to the truncated length, as then we won't know if it was truncated or if the string's size happens to be exactly that of the buffer.

```cpp
if (n > (int) sizeof(buf) - 1) {
    dmLogError("Buffer underflow when formatting message-error (LogMessageError)");
}
```

The original code expected `dmSnPrintf` to have returned the untruncated length, which would have made it all work. Here, I get the same result by manually setting `n` to a length greater than what the buffer can hold. Alternatively, I could instead add a `bool is_truncated`, set it here, and check it in the condition below; it would mean more changed lines though.